juriscraper/pacer/appellate_attachment_page.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import print_function
+
+import pprint
+import re
+import sys
+
+from lxml import html
+
+from juriscraper.lib.html_utils import strip_bad_html_tags_insecure
+from juriscraper.lib.string_utils import force_unicode
+from .reports import BaseReport
+from ..lib.log_tools import make_default_logger
+
+logger = make_default_logger()
+
+
+class AppellateAttachmentPage(BaseReport):
+    """An object for querying and parsing the appellate att. page report."""
+
+    PATH = "doc1/"
+
+    def __init__(self, court_id, pacer_session=None):
+        super(AppellateAttachmentPage, self).__init__(court_id, pacer_session)
+        # Note that parsing bankruptcy attachment pages does not reveal the
+        # document number, only the attachment numbers.
+        self.is_bankruptcy = self.court_id.endswith("b")
+
+    def query(self, document_number):
+        """Query the "attachment page" endpoint and set the results to self.response.
+
+        :param document_number: The internal PACER document ID for the item.
+        :return: a request response object
+        """
+        assert (
+            self.session is not None
+        ), "session attribute of DocketReport cannot be None."
+        # coerce the fourth digit of the document number to 1 to ensure we get
+        # the attachment page.
+        document_number = document_number[:3] + "0" + document_number[4:]
+        url = self.url + document_number
+        logger.info(u"Querying the attachment page endpoint at URL: %s", url)
+        self.response = self.session.get(url)
+        self.parse()
+
+    def _strip_bad_html_tags_insecure(self, tree):
+        """Remove bad tags from HTML while leaving scripts.
+
+        :param tree: A tree you wish to cleanup
+        :type tree: lxml.html.HtmlElement
+        Set self.tree to a cleaned lxml.html.HtmlElement
+        :return: None
+        """
+        self.tree = strip_bad_html_tags_insecure(tree, remove_scripts=False)
+
+    @property
+    def data(self):
+        """Get data back from the query for the matching document entry.
+
+        Appellate attachments is different than the district court.  We have
+        no document_id. Instead we can use the pacer_seq_no or the dls_id
+        to discern which attachment goes with which docket entry.
```

Comment:
Some appellate court have document numbers.

Reply:
@johnhawkinson could you point me to some examples to flush out tests and the code?

Reply:
This is fixed - my reading of the document numbers and the "dls_id" was in error.